d2/src/main/proto/XdsD2.proto

```diff
@@ -40,5 +40,55 @@ message D2SymlinkNode {
 }
 
 message D2NodeMap {
+  // Deprecated in favor of D2UriMap
+  option deprecated = true;
   map<string, D2Node> nodes = 1;
 }
+
+// D2URI is a proto representation of com.linkedin.d2.balancer.properties.UriProperties. Note that a D2 UriProperties is
+// actually an announcement from a single machine for potentially multiple URIs. This class is a misnomer but it seems
+// important to be consistent with the old naming strategy as this new proto bridges the old type. The string keys in
+// the maps of the message below will be the URIs of this announcement. Here is a sample ZK announcement for additional
+// clarity:
+//	{
+//		"weights": {
+//			"https://foo.stg.linkedin.com:18792/Toki/resources": 1.0
+//		},
+//		"partitionDesc": {
+//			"https://foo.stg.linkedin.com:18792/Toki/resources": {
+//				"0": {
+//					"weight": 1.0
+//				}
+//			}
+//		},
+//		"uriSpecificProperties": {
+//			"https://foo.stg.linkedin.com:18792/Toki/resources": {
+//				"com.linkedin.app.version": "0.1.76"
+//			}
+//		},
+//		"clusterName": "Toki"
+//	}
+message D2URI {
+  // The ZK stat of the node that contained this announcement.
+  Stat stat = 1;
+
+  // The name of the cluster this announcement belongs to. This is inferred from the original "clusterName" field.
+  string cluster_name = 2;
+
+  message PartitionData {
+    // Maps the partition number to the desired D2 weight for that partition.
+    map<int32, double> weights = 1;
+  }
+  // The partitions for each URI contained in this announcement. This is inferred from the original "partitionDesc" and
+  // "weights" fields. If "partitionDesc" is present in the original ZK node, it is always used regardless of "weights".
+  // Otherwise, "weights" is assumed to be for partition 0, as specified in UriProperties.
+  map<string, PartitionData> partition_desc = 3;
+
+  // Additional metadata for each URI in this announcement. This is inferred from the original "uriSpecificProperties"
+  // fields.
+  map<string, google.protobuf.Struct> uri_specific_properties = 4;
```

Comment:
This is `Map<URI, Map<String, Object>>` in the java code. Can we similarly make it `map<string, map<string, struct>` here? You may need to declare the nested map separately though.

Reply:
`map<string, struct>` is effectively the same as `struct`, isn't it?